File: app2/src/main/java/com/ssz/framejava/app/AppContext.java
package com.ssz.framejava.app;

import android.app.Application;

import com.ssz.framejava.base.app.func.IApp;
import com.ssz.framejava.base.app.helper.AppHelper;
import com.ssz.framejava.base.ui.dagger.di.component.AppComponent;
import com.ssz.framejava.base.ui.dagger.di.component.DaggerAppComponent;
import com.ssz.framejava.base.ui.dagger.di.module.AppModule;
import com.ssz.framejava.base.ui.dagger.di.module.NetModule;
import com.ssz.framejava.utils.log.TimberUtil;


/**
 * @author : zsp
 * time : 2019 11 2019/11/1 16:31
 */
public final class AppContext extends Application implements IApp {

    @Override
    public void onCreate() {
        super.onCreate();
        TimberUtil.init();
        AppHelper.init(this).toLog();
        // Dagger2 全局配置
        setupAppComponent();
    }

    /**
     * Dagger2 support
     */
    private AppComponent appComponent;
    public AppComponent getAppComponent(){
        return appComponent;
    }
    @Override
    public void setupAppComponent() {
        appComponent = DaggerAppComponent.builder()
                .appModule(new AppModule(this))
                .netModule(new NetModule())
                .build();
        appComponent.inject(this);
    }
}

File: app/src/main/java/com/ssz/studydemo/app/AppContext.kt
package com.ssz.studydemo.app

import android.app.Application
import com.ssz.studydemo.base.app.func.IApp
import com.ssz.studydemo.base.app.helper.AppHelper
import com.ssz.studydemo.base.ui.dagger.di.component.AppComponent
import com.ssz.studydemo.base.ui.dagger.di.component.DaggerAppComponent
import com.ssz.studydemo.base.ui.dagger.di.module.AppModule
import com.ssz.studydemo.base.ui.dagger.di.module.NetModule
import com.ssz.studydemo.utils.network.NetworkManager

/**
 * @author : zsp
 * time : 2019 12 2019/12/17 10:04
 */
class AppContext : Application(), IApp {

    var ji = "我在 master 里 加了个成员变量"

    override fun onCreate() {
        super.onCreate()
        AppHelper.init(this).toLog()
        setupAppComponent()
        // 网络状态监听框架注册
        NetworkManager.getDefault().init(this)
    }

    /**
     * supperDagger2
     */
    override lateinit var mAppComponent: AppComponent
    override fun setupAppComponent() {
        mAppComponent = DaggerAppComponent.builder()
                .appModule(AppModule(this))
                .netModule(NetModule())
                .build()
        mAppComponent.inject(this)
    }
}